Add Quotas API to Kafka CR and upgrade Strimzi Quotas Plugin to 0.3.1 (#9895)
strimzi/strimzi-kafka-operator: documentation/modules/managing/proc-setting-broker-limits.adoc

For example, you can set a byte-rate threshold of 40 MBps for producers.
If two producers are running, they are each limited to a throughput of 20 MBps.

> Storage quotas enforce throttling on Kafka disk storage when it reaches the specified limit.

Reviewer: ```suggestion
Storage quotas enforce the throttling of Kafka producers based on Kafka disk storageutilization when it reaches the specified limit.
```

You can specify the limit in bytes or percentage of available disk space.
The limit applies to all available disk space across all brokers.

> It prevents disks filling up too quickly and exceeding their capacity.

Reviewer: I think this is misleading. The limits apply to every disk individually. No `accross all brokers` or against `available disk space`.

spec:
  kafka:
    # ...
    quotas:
      type: strimzi

>       producerByteRate: 1000000 <1>
>       consumerByteRate: 1000000 <2>
>       minAvailableBytesPerVolume: 500000000000 <3>
>       excludedPrincipals: <4>

Reviewer 2: ```suggestion
      producerByteRate: 1000000 # <1>
      consumerByteRate: 1000000 # <2>
      minAvailableBytesPerVolume: 500000000000 # <3>
      excludedPrincipals: # <4>
```

        - my-user
----
<1> Sets the producer byte-rate threshold. 1 MBps in this example.
<2> Sets the consumer byte-rate threshold. 1 MBps in this example.
<3> Sets the available bytes limit for storage. 500 GB in this example.
<4> Sets the list of excluded users that are removed from the quota. `my-user` in this example.

. Update the resource.
+
[source,shell,subs=+quotes]

> kubectl apply -f _<kafka_configuration_file>_

Author: So you would remove the
```
. Update the resource.
+
[source,shell,subs=+quotes]
kubectl apply -f _<kafka_configuration_file>_
```
part? Or just replace the `Update the resource` with `Apply the changes ...`?

> kubectl apply -f _<kafka_configuration_file>_

Reviewer 2: Yes, remove that. The step can be just: ". Apply the changes to the `Kafka` configuration."

> NOTE: `minAvailableBytesPerVolume` and `minAvailableRatioPerVolume` are mutually exclusive, you can configure
> only one of them.

Reviewer 2: ```suggestion
NOTE: `minAvailableBytesPerVolume` and `minAvailableRatioPerVolume` are mutually exclusive. This means that only one of these parameters should be configured.
```
